.github/workflows/build.yml

```diff
@@ -0,0 +1,144 @@
+          echo "::set-output name=VARIANT_FHEM::$(echo "$VARIANT_FHEM")"
+          echo "::set-output name=BUILD_DATE::$(  date --iso-8601=seconds --utc )"
+          # detect rolling branch
+          if [[ ${{ steps.gitVars.outputs.BRANCH }} == "master" ]]; then
+          echo "::set-output name=TAG_ROLLING::latest"
+          else
+          echo "::set-output name=TAG_ROLLING::$( echo "${{ steps.gitVars.outputs.BRANCH }}" )"
+          fi
+
+      - name: Check variables
+        run: |
+          echo FHEM_REVISION_LATEST: ${{ steps.svnVars.outputs.FHEM_REVISION_LATEST }}
+          echo FHEM_VERSION: ${{ steps.svnVars.outputs.FHEM_VERSION }}
+          echo IMAGE_VERSION: ${{ steps.gitVars.outputs.IMAGE_VERSION }}
+          echo BUILD_DATE: ${{ steps.buildVars.outputs.BUILD_DATE }}
+          echo BRANCH: ${{ steps.gitVars.outputs.BRANCH }}
+          echo VARIANT: ${{ steps.buildVars.outputs.VARIANT }}
+          echo VARIANT_IMAGE: ${{ steps.buildVars.outputs.VARIANT_IMAGE }}
+          echo VARIANT_FHEM: ${{ steps.buildVars.outputs.VARIANT_FHEM }}
+          echo VAROLLING_TAG: ${{ steps.buildVars.outputs.TAG_ROLLING }}
+          
+      - name: Docker Setup QEMU
+        uses: docker/setup-qemu-action@v1.0.1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          load: false  
+          file: ./Dockerfile
+          platforms: linux/386,linux/amd64,linux/arm/v6,linux/arm/v7,linux/arm64,
+          # platforms: linux/arm/v6,linux/arm/v7,linux/amd64
+          push: false
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          tags: fhem:${{ steps.buildVars.outputs.VARIANT }}
+          build-args: |
+            PLATFORM=linux
+            BUILD_DATE=${{ steps.buildVars.outputs.BUILD_DATE }}
+            TAG=${{ steps.buildVars.outputs.VARIANT }}
+            TAG_ROLLING=${{ steps.buildVars.outputs.TAG_ROLLING }}
+            IMAGE_VERSION=${{ steps.buildVars.outputs.VARIANT }}
+            IMAGE_VCS_REF=${{ github.sha }}
+            FHEM_VERSION=${{ steps.buildVars.outputs.VARIANT_FHEM }}
+            VCS_REF=${{ steps.svnVars.outputs.FHEM_REVISION_LATEST }}
+            IMAGE_LAYER_SYS_EXT=0
+            IMAGE_LAYER_PERL_EXT=1
+            IMAGE_LAYER_DEV=0
+            IMAGE_LAYER_PERL_CPAN=1
+            IMAGE_LAYER_PERL_CPAN_EXT=0
+            IMAGE_LAYER_PYTHON=0
+            IMAGE_LAYER_PYTHON_EXT=0
+            IMAGE_LAYER_NODEJS=0
+            IMAGE_LAYER_NODEJS_EXT=0
```

Comment:
Please don't exclude any of the layers as it will otherwise break existing configurations.
All layers are essentially needed for maximum compatibility with existing FHEM modules.

Reply:
1)
I excluded the layers, because espacally nodejs 14 isn't supported on some of the platforms anymore and this results in different behaviours of the images between platforms.
I am aware of, that this is may be a breaking change for some special cases. But for this cases it is still possible to install the needed packages via docker argument -E.

To save some time during testing an get some minmal working image i decided to disable all those packages to get a pure fhem image which can run fhem similar to the debian package which also has no dependencys to nodejs, python and so on. This redudces also some complexity during setup of the gighub workflow.
Currently the images aren't tested because they are nowhere published. In my oppinion it's best to first check if the base image works on the desired platforms and then move on to extend them again.

For nodejs there are also already available images which may can run the needed services for thos fhem modules who communitcate with a nodejs service . Also the health-check.sh isn't aware of the nodejs services or other laayer, so may it is better to move them to a separate service which is maintained seperate.

IMAGE_LAYER_DEV seems to bring in tools to compile software, which should normaly be not be included in a normal d docker image. Also all this stuff can be added via -E argument if really needed.

Maybe it's worth of providing some -minimal and some -full image variants, depending on the platform. But for now, my goal was to build something of a base image.

Reply:
The different layers for platforms are intentionally to provide maximum possibility on each platform. Providing even more variants is not beneficial in terms of increased maintenance and support effort.

Build tools in the image are required to support FHEM Installer and related components to install Perl modules and other stuff automatically.

Reply:
If the build tools are only needed to install the packages into the image, we can remove them after the installation process is done in a seperate layer.

There is also a job to create a image which has all that services included in the image, same as before. But i think that's not the docker way to have services in different containers.